### PR TITLE
Study@3

### PR DESCRIPTION
**Pull request by jonathanong**
closes https://github.com/dollarshaveclub/study/issues/20

primary changes:

- removes default stores
- tests on jsdom
- adds `default` and `winner` support
- fixes a className bug

### CODE REVIEW

File: src/study.js [hunk 1]
```diff
@@ -1,137 +1,114 @@
...
-        classesToRemove.forEach(className => document.body.classList.remove(className));
-        if (this.classList[name]) document.body.classList.add(`${name}--${this.classList[name]}`);
-      });
-      document.removeEventListener('DOMContentLoaded', applyClasses);
-    };
+  removeClasses(testName, exceptClassName) {
+    const { root } = this;
+    if (!root) return;
+
+
```
Comment (briangonzalez) [inline src/study.js]: remove line

File: src/study.js [hunk 2]
```diff
@@ -1,137 +1,114 @@
...
+    const currentClassNames = root.className.split(/\s+/g)
```
Comment (briangonzalez) [inline src/study.js]: Could we use [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) here instead?

File: src/study.js [hunk 3]
```diff
@@ -1,137 +1,114 @@
...
-   * info such as name and buckets
-   */
-  define(data) {
-    let normalizedData = data;
-    if (!Array.isArray(data)) { normalizedData = [data]; }
+  define(tests) {
+    let normalizedData = tests;
+    if (!Array.isArray(tests)) normalizedData = [tests];
 
     normalizedData.forEach((test) => {
```
Comment (briangonzalez) [inline src/study.js]: remove parens

File: src/study.js [hunk 4]
```diff
@@ -1,137 +1,114 @@
...
+  }
+
+  assignAll() {
+    const {
+      previousAssignments,
+      userAssignments,
+      persistedUserAssignments,
+    } = this;
 
     this.providedTests.forEach((test) => {
```
Comment (briangonzalez) [inline src/study.js]: remove parens

File: src/study.js [hunk 5]
```diff
@@ -142,86 +119,38 @@ class Study {
...
+  assign(testName, bucketName) {
+    if (!testName) return this.assignAll();
+
+    const test = this.providedTests.filter(x => x.name === testName)[0];
+    if (bucketName === null || !test) {
+      delete this.userAssignments[testName];
+      delete this.persistedUserAssignments[testName];
+      this.removeClasses(testName);
+    } else {
+      this.userAssignments[testName] =
```
Comment (briangonzalez) [inline src/study.js]: eh? where's the right hand side of this assignment?

Comment (jonathanong) [inline src/study.js]: right below it ;)

File: src/study.js (hunk 4 above)
Comment (jonathanong) [inline src/study.js]: airbnb eslint -_- we need to just make `eslint-config-dsc`

File: src/study.js (hunk 5 above)
Comment (briangonzalez) [inline src/study.js]: i would refactor this to be more readable. took a lot of my mental energy to grok what was going on here.

Comment (jonathanong) [inline src/study.js]: hahaha i can make it one line instead

File: src/study.js (hunk 2 above)
Comment (jonathanong) [inline src/study.js]: it doesn't support returning all classes :/

File: src/stores/memory.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+
+
```
Comment (briangonzalez) [inline src/stores/memory.js]: Remove new line

**Pull request merged**